condensation-forum/application.py

```diff
@@ -301,6 +314,117 @@ def logoutCallback():
     print("User signed out: " + user["name"], file=sys.stderr)
 
 
+@application.route('/file-delete', methods=['POST'])
+@authManager.requireAuthentication
+def fileManagerDeleteHander():
+    user = authManager.getUserData()
+    fid = int(request.form['file'])
+    
+    if not user:
+        abort(403)
+    id = user['id']
+
+    #delete the file by Cloud_key in AWS S3
+    with dataSessionMgr.session_scope() as dbSession:
+        file1 = query.getFileById(dbSession,fid)
+        file1 = query.extractOutput(file1)
+    
+    key = file1['cloud_key']
+    try:
+        s3client.delete_object(Bucket=bucket_name,Key=key)
+    except Exception as e:
+        print("Something happen: ",e)
+        return e
+    
+    #delete the file by fileID in DB
+    with dataSessionMgr.session_scope() as dbSession:
+        file = query.getFileById(dbSession,fid)
+        if file:
+            dbSession.delete(file)
+            flash("File Deleted")
+  
+    return redirect(url_for("fileManager"))
+
+@application.route('/fileManager', methods=['GET', 'POST'])
+@authManager.requireAuthentication
+def fileManager():
+    user = authManager.getUserData()
+    if request.method == 'POST':
+        # do stuff when the form is submitted
+        if not user:
+            abort(403)
+        id = user['id']
+
+        if 'file' not in request.files:
+            flash('No file part')
```

Comment:
What does "no file part" mean? what could cause this error, and how can you present information to the user to help them correct it?

Reply:
Removed. We already have the flash if no file in request..files